### DISCUSSION TITLE
How to configure Win7 to use vintage NI gear with nimi-Python

### DISCUSSION THREAD

**Discussion by dbbotkin** [ni/nimi-python#1814] [Show and tell]
DISCLAIMER: I don't think 'obsolete' electronic gear belongs in the dump; put it to use learning Python instrumentation.

Being a 'vintage' engineer myself, I find pleasure in refurbishing top quality NI electronics for hobby use. The challenge is in the software setup. NI has a 'Community' program for folks who are into instrumentation just for the fun of it. Amateur radio or just helping kids with mechatronics projects. I tried Windows 10 at first as that is the platform for the Community program. Great software for the latest gear; older instruments are not always supported, so I rolled back to Win7.  I'm mostly MacOS these days, but NI isn't. Linux is a rabbit hole; some like that.

Here are the key points:
1) You will need Win7 with service pack 1, preferably 64bit (more later)
2) Start with PXI modules and chassis, as they are more available on eBay than are PXIe.
3) Get a PXI-8105 controller (32bit)
4) Later . . . (if you are good with ESD, etc.) Make the 8105 into a 64bit racehorse by swapping the CPU chip from stock T2500 to a T7400 as in the 8106. T7400's are available in quantity from overseas for under $20. A boost in speed as well. It's a 30 minute task & simple tools.

Now that's what I did, there are other paths I'm sure. You will need the 64bit to get 18.1 level NI software as that is the optimal for a Win7 system. Going higher means Package Manager and maybe your gear won't make the cut, just sayin'.

If you are already setup with Python, great! If you are just starting out, try Thonny as the interface as it comes with the most common Python version installed and there is no need to get into a command line terminal or figure out repositories. I downloaded the nimi-python modules with little more than a keystroke and got a PXI-4071 DMM capturing data right away. Some modules might be more difficult, nidcpower for example. NI SMU's are more complex than a PSU/DMM combo; a lot more.

**Comment by marcoskirsch** [comment]
> Great software for the latest gear; older instruments are not always supported, so I rolled back to Win7. 

What instruments do you have that don't work with anything newer than Windows 7?
Supporting Windows 7 is not as simple as it sounds anymore.

> I'm mostly MacOS these days, but NI isn't.

I use a Mac myself, and have since the mid eighties. I wish the business case for NI to support its instrumentation drivers on Mac was there... but alas that's not the case and the ever-shifting driver model in OS doesn't help.

**Reply by dbbotkin** [reply]
Instruments:
I started out with LabVIEW maybe fifteen years ago with Arduino and some of MyDAQ gear. In 2008 I got more involved with electronics with my grandson and LabVIEW 2009 as he was very interested in science at the time. That led to a process of finding stuff on eBay that could be made useful with a little cleaning and adjustment. The my GF, now wife suggested that we could put the electronics to work helping kids with mechatrics projects. So some of the early acquisitions that might have problems would be a 4060 DMM, 5401 FGEN, an 8.5 digit DMM from Signametrics (now Keysight), a scope from LeCroy. Those will work with XP. Now there is a lot more gear that works ok with Win7--I never considered Vista and didn't know about Win 8 as I was deep in MacOS when it came out.

MacOS:
I've been  a fanboy since the beginning, but always had a 'business' need for a Windows system so I'm pretty adept at both. MacOS is a far better platform for getting things done without being distracted by 'features" that seem to have found a home in Win10. Started using remote desktop (or the =) from a Mac to control screens of PC's and other Macs, mostly doing scientific/finance calculations, etc. Now a MacOS with Microsoft Remote Desktop is my SOP. I have an old Intel iMac that runs Windows 10 (unless some MSFT update kills it. Also a maxed out PC that I built for running NI Community edition software. The bloatware in Win10 was a distraction, so I ran a cleanup.

**Reply by dbbotkin** [reply]
UPDATE:
I have started testing Windows 10 on one of my PXI-8105 Controllers. It is the first to get an upgrade to the T7400 CPU that supports 64bit code and it was working just fine with modules, so I checked the specs for Win10 and found--to my surprise--it more than qualified. I bought W10 Pro on Amazon and got an actual COA and sticker with a real product key. For $150+. I installed direct to SSD 500GB--needed to delete stock partitions and make new for install--and an hour or so later, it was ready to load NI MAX and friends.

Bloatware. That is what makes Win10 a miserable experience coming from MacOS and Win7. If you are a gamer, into social media or still living with Mom, you might like the eye candy and 'features'; I did not, so 'delete' was my fave button. HINT: delete the bloatware **before** NI loading.

Am I happy? It took a while. Because of the security features in Win10 and the clever way forms exactly like the ones from XP were hidden in 'administrative tools'. In particular, avoiding the 'Edge' and getting Remote Desktop going were _sine qua non_.

After 24hours and being able to use a big screen--as well as supervising NI loading on an iPad--I agree; 10 is better than 7 and using more recent NI software with Python is more than worth the $150.
